guides/auth-torii-with-github.md

```diff
@@ -0,0 +1,581 @@
+
+We added the configuration for our authentication, but we didn't yet add the pieces that use it. Let's do that first.
+
+```
+ember g authenticator torii
+```
+
+and we'll modify our application and routes with some useful mixins.
+
+First, we'll add the [ApplicationRouteMixin](http://ember-simple-auth.com/api/classes/ApplicationRouteMixin.html) to
+our application route. This is optional, but adds methods supporting the authentication lifecycle that we would
+otherwise have to implement explicitly.
+
+```js
+import Ember from 'ember';
+import ApplicationRouteMixin from 'ember-simple-auth/mixins/application-route-mixin';
+
+export default Ember.Route.extend(ApplicationRouteMixin, {
+});
+```
+
+Next, we'll designate the `index` route as an authenticated route using the
+[AuthenticatedRouteMixin](http://ember-simple-auth.com/api/classes/AuthenticatedRouteMixin.html). This will make the
+route inaccessible until we finish the authentication. It will automatically redirect you to the specified login route,
+by default `login`, if you are not authenticated.
+
+```js
+// app/routes/index.js
+
+import Ember from 'ember';
+import AuthenticatedRouteMixin from 'ember-simple-auth/mixins/authenticated-route-mixin';
+
+export default Ember.Route.extend(AuthenticatedRouteMixin, {
+});
+```
+
+If you're still running the app when you save this, you will see it redirect to the `login` route.
+
+We'll also designate the `login` route as available for unauthenticated access by applying the
+[UnauthenticatedRouteMixin](http://ember-simple-auth.com/api/classes/UnauthenticatedRouteMixin.html) to it. This will
+redirect you to the `routeIfAlreadyAuthenticated` which defaults to `index`. As you can see, `ember-simple-auth` has
+sensible and convenient defaults.
+
+Next, we'll set up our torii authenticator to start.
+
+```js
+// app/authenticators/torii.js
+
+import Ember from 'ember';
+import ToriiAuthenticator from 'ember-simple-auth/authenticators/torii';
+
+export default ToriiAuthenticator.extend({
+  torii: Ember.inject.service()
+});
+```
+
+We also need to define a Torii provider. Because Torii doesn't provide a generator, we need to create the `app/torii-providers`
+directory ourselves and create the following `github.js` inside it.
+
+```js
+// app/torii-providers/github.js
+
+import GitHubOAuth2Provider from 'torii/providers/github-oauth2';
+
+export default GitHubOAuth2Provider.extend({
+  fetch(data) {
+    return data;
+  }
+});
+```
+
+There's only one more piece in this step, connecting our "Log in" button to the authentication mechanism. First, add an
+action to your `login` route.
+
+```js
+  // add to app/routes/login.js
+  
+  actions: {
+    login() {
+      this.get('session').authenticate('authenticator:torii', 'github');
+    }
+  }
+```
+
+Finally, change your `login` template to send the action when the button is pressed.
+
+```handlebars
+{{!-- app/templates/login.hbs --}}
+
+<button {{action "login"}}>Log in to GitHub</button>
```

Comment:
We should use modern action in this guide, helping make people familiar with modern/idiomatic Ember concepts, so this would be:

```hbs
<button onclick={{action "login"}}>Log in to GitHub</button>
```

and the above implementation of the action would go to the controller.